### DISCUSSION TITLE
New package: pocketbase-presigned-urls

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#6001] [Show and tell]
This JSVM package will serve presigned file URLs directly to your s3 storage source instead of serving them through your PocketBase instance. 

https://www.npmjs.com/package/pocketbase-presigned-urls

Features:
* Works with PocketBase file API security
* Lower latency and fewer hops required to serve the bulk of the file data
* Possible savings on bandwidth when using providers like iDrive E2 and Tigris who provide free egress
* Compatible with all versions of PocketBase

Thanks to @mmgfrcs for contributing. Please try it out and let me know if it's working!

**Comment by thangngoc89** [comment]
@benallfree just FYI, I implemented this at a Golang hook [a little while ago](https://github.com/pocketbase/pocketbase/discussions/4871) for the exact reason. I'm using Cloudflare R2 so free egress fee.

**Reply by benallfree** [reply]
Thank you! This may come in handy in my new project.

(Tigris has free egress too.)

**Reply by thangngoc89** [reply]
@gedw99 not sure where did you get that information. While Tigris isn't the cheapest in term of storage cost (Cloudflare R2 is cheaper at $0.015/GB), it's definitely cheaper than Amazon S3 (us-east-1), $0.023/GB
